### ISSUE TITLE
Site Title: prevent from saving and rendering a value made of only spaces

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60467]
### Description

Empty links are far from ideal for SEO and accessibility.

Turns out that:
- The classic admin doesn't allow to save a Site Title value made of only paces.
- The editor Site Title block does.

So that there is an inconsistency in the way the site title value is first of all _saved_ to the database.
This inconsistency may lead to rendering an empty link on the front end.

### Step-by-step reproduction instructions

- Use a theme that has the Site Title block on one of its templates e.g. Twenty Twent-Four > Blog Home template.
- Go to the classic admin General Settings > Site Title
- Empty the site title value and save.
- In a new browser tab, go to the front end home page.
- Observe the site title is not rendered. 
- Switch to the browser tab with the classic admin General Settings page.
- Enter two spaces in the site title input field and save.
- After the page refresn, observe the site title input field does not contain spaces. It's empty, which proves classic WordPress doesn't allow to save a value of only spaces.
- Keep this tab open, and open a new tab to go to the Site Editor.
- Select the Site Title block, enter two spaces and save.
- Switch to the browser tab with the classic admin General Settings page and refresh the page.
- Observe the site title input field now does contain two spaces.
- In a new browser tab, go to the front end home page.
- Observe the Site Title block is now rendered but the link contains two spaces. This is an empty link, far from ideal for SEO and accessibility and should be avoided. Example of the rendered markup:

```
<p class="wp-block-site-title">
    <a href="http://localhost:8889" target="_self" rel="home" aria-current="page">   </a>
</p>
```

- Switch to the browser tab with the classic admin General Settings page.
- Don't do any edit and save.
- After the page refreshes, observ ethe two spaces in the site titel input field have been removed.
- Go to the front end and observe the empty site title is not rendered any longer.
- Observe [the block render callback](https://github.com/WordPress/gutenberg/blob/9c6d4fe59763b188d27ad937c2f0daa39e4d9341/packages/block-library/src/site-title/index.php#L17-L21) does check whether a site title value is set, but it doesn't trim it.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Note: I created a Trac ticket for a similar issue with the Site _Logo_ but that's more responsibility of Core. See https://core.trac.wordpress.org/ticket/60922#ticket
